### PR TITLE
WIP: shouldi_subflows: subflow for js packages

### PR DESCRIPTION
**Pull request by Yash-Varshney**

### CODE REVIEW

**Comment by Yash-Varshney** [comment]
@pdxjohnny made changes for running the JS package.  Pls, guide further !!

File: examples/shouldi/shouldi/cli.py [hunk 1]
```diff
@@ -182,12 +183,67 @@ async def is_lang_javascript(self, repo):
...
+@op(
+    inputs={"repo": clone_git_repo.op.outputs["repo"]},
+    outputs={"result": SA_RESULTS},
+    conditions=[is_lang_javascript.op.outputs["javascript"]],
+)
+async def run_javascript_sa(self, repo):
+    """
+    Run JS static analysis
+    """
+    setup_kwargs = SetupPyKWArg.get_kwargs(
```
Comment (pdxjohnny) [inline examples/shouldi/shouldi/cli.py:213]: This is a CLI util that can be also used programaticlly to extract information from setup.py files, but only for Python setup.py files. https://intel.github.io/dffml/master/cli.html#kwarg

You won't need this here

File: examples/shouldi/shouldi/cli.py [hunk 2]
```diff
@@ -182,12 +183,67 @@ async def is_lang_javascript(self, repo):
...
+        os.path.join(repo.directory, "package.json")
+    )
+
+    # TODO Make is so that
+    async with self.octx.parent(DATAFLOW_SA_JAVASCRIPT) as octx:
+        async for _, results in octx.run(
+            [
+                Input(
+                    value=setup_kwargs["name"],
+                    definition=pypi_package_json.op.inputs["package"],
```
Comment (pdxjohnny) [inline examples/shouldi/shouldi/cli.py:223]: Here you'll want to have the input be the input for `run_npm_audit`, so you'll want to make the definition, `run_npm_audit.op.inputs["pkg"]`, the value would be `repo.directory`. So this gives the dataflow an input which it sees should go to `run_npm_audit` which is going to use that as it's `pkg` argument.

File: CHANGELOG.md [hunk 3]
```diff
@@ -13,6 +13,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - shouldi got an operation to run Dependency-check on java code.
 - `load` and `run` functions in high level API
 - Doctestable examples to `db` operations.
+- shouldi_subflows: added subflow for running JavaScript pkg.
```
Comment (pdxjohnny) [inline CHANGELOG.md:16]:
```suggestion
```

File: examples/shouldi/shouldi/cli.py [hunk 4]
```diff
@@ -182,12 +183,56 @@ async def is_lang_javascript(self, repo):
...
+    async with self.octx.parent(DATAFLOW_SA_JAVASCRIPT) as octx:
+        async for _, results in octx.run(
+            [
+                Input(
+                    value=repo.directory,
+                    definition=run_npm_audit.op.inputs["pkg"],
+                )
+            ]
+        ):
+            # TODO Make this report more useful
```
Comment (pdxjohnny) [inline examples/shouldi/shouldi/cli.py:216]:
```suggestion
```

File: examples/shouldi/shouldi/cli.py [hunk 5]
```diff
@@ -182,12 +183,56 @@ async def is_lang_javascript(self, repo):
...
+    inputs={"repo": clone_git_repo.op.outputs["repo"]},
+    outputs={"result": SA_RESULTS},
+    conditions=[is_lang_javascript.op.outputs["javascript"]],
+)
+async def run_javascript_sa(self, repo):
+    """
+    Run JS static analysis
+    """
+
+    # TODO Make is so that
```
Comment (pdxjohnny) [inline examples/shouldi/shouldi/cli.py:206]:
```suggestion
```

File: examples/shouldi/tests/test_cli.py [hunk 6]
```diff
@@ -20,3 +20,18 @@ async def test_install(self):
             output = stdout.getvalue()
         self.assertIn("shouldi is okay to install", output)
         self.assertIn("Do not install insecure-package!", output)
+
+    async def test_install_js(self):
+        import pathlib
```
Comment (pdxjohnny) [inline examples/shouldi/tests/test_cli.py:25]: Please clean up these testcases and make sure they are really checking the results

Comment (pdxjohnny) [reply examples/shouldi/tests/test_cli.py:25]: You may need to modify cli.py

Comment (Yash-Varshney) [reply examples/shouldi/tests/test_cli.py:25]: Sorry , I didn't quite get it. 
On running -
`$ cd examples/shouldi
$ python setup.py test -s tests.test_cli.TestCLI.test_install_js`
It shows the accurate results.

File: examples/shouldi/tests/test_cli.py [hunk 7]
```diff
@@ -20,3 +20,18 @@ async def test_install(self):
...
+
+        await ShouldI.install.cli(
+            "https://github.com/trekhleb/javascript-algorithms"
+        )
+
+        return
```
Comment (pdxjohnny) [inline examples/shouldi/tests/test_cli.py:31]: The code after this statement is similar to what you'll want the testcase to be

**Pull request closed without merging**